ResearchKit/Consent/ORKVisualConsentStepViewController.m

```diff
@@ -251,10 +252,30 @@ - (IBAction)goToPreviousPage {
 }
 
 - (IBAction)next {
-    [self showViewController:[self viewControllerForIndex:[self currentIndex]+1] forward:YES animated:YES];
-    ORKAccessibilityPostNotificationAfterDelay(UIAccessibilityScreenChangedNotification, nil, 0.5);
+    ORKConsentSceneViewController *currentConsentSceneViewController = [self viewControllerForIndex:[self currentIndex]];
+    [currentConsentSceneViewController setScrollEnabled:NO];
+    if ([currentConsentSceneViewController isScrolledToTop])
+    {
+        [self showNextViewController];
+    }
+    else
+    {
+        [currentConsentSceneViewController setScrollViewDelegate:self];
+        [currentConsentSceneViewController scrollToTopAnimated:YES];
```

Comment:
This is hacky. Might be better to change the API to the consent scene to animate with a completion, and do a direct animation on the bounds rather than setContentOffset:animated:. Then you'll get a block callback on the animation completion, whether the animation actually finishes or not, which is also less fragile.